### DISCUSSION TITLE
how to transform generated type?

### DISCUSSION THREAD

**Discussion by well-balanced** [prisma/prisma#14412] [Q&A] [answered]
hi, thank you always.
I was using JSON data type on the `props` field. so the type of `props` field is `Prisma.JsonValue`.
I had no choice but to use the type assertion to create another object using the type I had received.

e.g.
```ts
const instance = await prisma.someModel.findUnique({ where: { id } })
// Types of property 'props' are incompatible. 
// Type 'JsonValue' is not assignable to type 'InputJsonValue'. 
// Type 'null' is not assignable to type 'InputJsonValue'.ts
await prisma.someModel.create({ data: { props: instance.props } })

await prisma.someModel.create({ data: { props: instance.props as Prisma.InputJsonValue } }) // no error
```

of course, to solve this problem, I can use the method of checking null using the if statements. but I'm looking for another solution. 

```ts
namespace Prisma {
  ...
  export type JsonValue = string | number | boolean | JsonObject | JsonArray | null
  export type InputJsonValue = string | number | boolean | InputJsonObject | InputJsonArray
  ...
}
```

I just want to exclude `null` type from the `JsonValue` type.
do you have any API or solution?

thanks :)

**Comment by pantharshit00** [comment, marked as answer]
Hello @well-balanced, our apologies for the delay in getting back to you! Due to the large number of questions in our community channels, unfortunately this slipped through the cracks. 

Right now we don't have any solution for this. Maybe you can use something like [patch-package](https://github.com/ds300/patch-package) here. I think you should open a feature request for this: https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
